crates/ruff/src/args.rs

```diff
@@ -812,14 +812,24 @@ impl TypedValueParser for ConfigArgumentParser {
         arg: Option<&clap::Arg>,
         value: &std::ffi::OsStr,
     ) -> Result<Self::Value, clap::Error> {
-        let path_to_config_file = PathBuf::from(value);
-        if path_to_config_file.exists() {
-            return Ok(SingleConfigArgument::FilePath(path_to_config_file));
-        }
+        // Convert to UTF-8.
+        let Some(value) = value.to_str() else {
+            // But respect non-UTF-8 paths.
```

Comment:
Worth adding a test with a non-UTF-8 path?

Reply:
I'm gonna defer it, I don't even know if Ruff as a whole will work with non-UTF-8 paths.